h2o-core/src/main/java/hex/Model.java

```diff
@@ -91,6 +91,10 @@ public boolean filter(KeySnapshot.KeyInfo k) {
     Frame scoreStagedPredictions(Frame frame, Key<Frame> destination_key);
   }
 
+  public interface Contributions {
+    Frame scoreContributions(Frame frame, Key<Frame> destination_key);
```

Comment:
I like the idea to have a custom key for the contribution matrix.

Reply:
In future, we have the possibility to add `Frame scoreContributions(Frame frame);` method, which will create the key.